e2e_test/source/cdc/cdc.share_stream.slt

```diff
@@ -96,6 +96,20 @@ create table orders_test (
    PRIMARY KEY (order_id)
 ) from mysql_mytest table 'mytest.orders';
 
+# Should not contain records inserted before the table is created (e.g. 'Bob' inserted above)
+statement ok
+create table orders_no_backfill (
+   order_id int,
+   order_date timestamp,
+   customer_name string,
+   price decimal,
+   product_id int,
+   order_status smallint,
+   PRIMARY KEY (order_id)
+) with (
+  snapshot = 'false'
+) from mysql_mytest table 'mytest.orders';
+
```

Comment:
Hmm I see that we insert `Sam` as a record, but where? Do you think it's possible for the cdc tests to be inlined?

Seems a little confusing because we create the table here (in `cdc.share_stream.slt`).

But we check for the contents in a separate file: (`e2e_test/source/cdc/cdc.check_new_rows.slt`).

Not sure how they are synced up.

Reply:
> Hmm I see that we insert Sam as a record, but where?

In the `e2e-source-test.sh` which is the "main" of e2e test. 

And `cdc.check_new_rows.slt` is called after cluster restart, which is to test cdc source can resume after recovery. I feel it is also odd to kill the cluster and restart it in the `.slt`.